resqpy/olio/relperm.py

```diff
@@ -0,0 +1,316 @@
+"""relperm.py: class for storing and retrieving dataframes of relative permeability
+data as RESQML objects.
+
+   note that this module uses the obj_Grid2dRepresentation class in a way that was not envisaged
+   when the RESQML standard was defined; software that does not use resqpy is unlikely to be
+   able to do much with data stored in this way
+"""
+import numpy as np
+import pandas as pd
+from scipy import interpolate
+import os
+import logging
+import resqpy.olio.xml_et as rqet
+from resqpy.olio.dataframe import DataFrame
+
+version = '27th July 2021'
+
+log = logging.getLogger(__name__)
+log.debug(f'dataframe.py version {version}')
+
+
+class RelPerm(DataFrame):
+   """Class for storing and retrieving a pandas dataframe of relative permeability data.
+
+    note:
+       inherits from DataFrame class
+    """
+
+   def __init__(self,
+                model,
+                uuid = None,
+                df = None,
+                uom_list = None,
+                realization = None,
+                phase_combo = None,
+                low_sal = False,
+                table_index = None,
+                title = 'relperm_table',
+                column_lookup_uuid = None,
+                uom_lookup_uuid = None):
+      """Create a new RelPerm object from either a previously stored property or a pandas dataframe.
+
+        arguments:
+           phase_combo (str, optional): the combination of phases whose relative
+           permeability behaviour is described. Options include 'water-oil', 'gas-oil' and
+           'gas-water'
+           low_sal (boolean, optional): if True, indicates that the water-oil table contains
+           the low-salinity data for relative permeability and capillary pressure
+           table_index (int, optional): the index of the relative permeability
+           table when multiple relative permeability tables are present. Note, indices should start at 1.
+
+        note:
+           see DataFrame class docstring for details of other arguments
+        """
+
+      # check that either a uuid OR dataframe has been provided
+      assert uuid is not None or df is not None, 'either a uuid or a dataframe must be provided'
```

Comment:
A minor suggestion - I think it would be more appropriate to raise a ValueError rather than an AssertionError in cases like this where the user-provided inputs are "wrong". Same for the other validation checks below.

`assert` is generally used to check for "internal" consistency, for ease of debugging and maintenance. AssertionErrors should generally not be thrown to a user unless there is a mistake in resqpy.

Reply:
Gotcha....I changed all the assert statements in the validation checks to if-else statements that raise ValueErrors